helm-charts/basehub/templates/dex/deployment.yaml

```diff
@@ -0,0 +1,82 @@
+{{- if .Values.dex.enabled -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: dex
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: dex
+  template:
+    metadata:
+      labels:
+        app: dex
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/dex/configmap.yaml") . | sha256sum }}
+        checksum/secret: {{ include (print $.Template.BasePath "/dex/secret.yaml") . | sha256sum }}
+    spec:
+      volumes:
+        - name: db
+          persistentVolumeClaim:
+            claimName: dex
+        - name: config
+          configMap:
+            name: dex
+      securityContext:
+        # The upstream repo runs with gid 0, and setting this makes
+        # sure the db volume we mount can be written to by the dex process
+        fsGroup: 0
+      containers:
+        - name: dex
+          image: ghcr.io/dexidp/dex:v2.32.0
+          ports:
+            - name: dex
+              containerPort: 5556
+          env:
+          # These are expanded by the dex config
+          - name: HUB_OAUTH2_CLIENT_SECRET
+            valueFrom:
+              secretKeyRef:
+                name: hub
+                key: hub.services.dex.apiToken
+          - name: OAUTH2_PROXY_CLIENT_SECRET
+            valueFrom:
+              secretKeyRef:
+                name: dex
+                key: oauth2Proxy.clientSecret
+          volumeMounts:
+          - name: config
+            mountPath: /srv/config
+          - name: db
+            mountPath: /srv/db
+          # Needs to be args, not cmd - this allows gomplate based
+          # expansion of config file
+          args:
+          - dex
+          - serve
+          - /srv/config/dex.yaml
+        - name: oauth2-proxy
+          image: quay.io/oauth2-proxy/oauth2-proxy:v7.3.0
+          command:
+          - oauth2-proxy
+          - --config=/srv/config/oauth2-proxy.cfg
+          volumeMounts:
+          - name: config
+            mountPath: /srv/config
+          ports:
+            - name: oauth2-proxy
+              containerPort: 9000
+          env:
+          # This is read by oauth2-proxy
+          - name: OAUTH2_PROXY_COOKIE_SECRET
+            valueFrom:
+              secretKeyRef:
+                name: dex
+                key: oauth2Proxy.cookieSecret
+          - name: OAUTH2_PROXY_CLIENT_SECRET
+            valueFrom:
+              secretKeyRef:
+                name: dex
+                key: oauth2Proxy.clientSecret
```

Comment:
Indentation nitpicking again.

Reply:
I personally prefer not indenting lists, but I also want to make these conversations go away asap so i'll just indent it :) Thank you for helping us keep things consistent, @consideRatio. Ideally an autoformatter would do this but I think those don't work on go template files like this.

Reply:
I re-read my comment and I have no idea if i prefer indenting lists or not indenting lists lol. I've just followed your suggestions now.